python/cudf/cudf/api/types.py

```diff
@@ -27,6 +27,35 @@
 )
 
 
+def dtype(arbitrary):
+    try:
+        np_dtype = np.dtype(arbitrary)
+        if np_dtype.name == "float16":
+            np_dtype = np.dtype("float32")
+        elif np_dtype.name in ("object", "str"):
+            np_dtype = np.dtype("object")
+        elif np_dtype.str == "<m8":
+            np_dtype = np.dtype("<m8[ns]")
+        elif np_dtype.str == "<M8":
+            np_dtype = np.dtype("<M8[ns]")
+        return np_dtype
```

Comment:
If someone does a `cudf.dtype('complex')`, I think we would end up returning `np.dtype('complex')` here, should we validate if the dtype exists in our cudf type map before returning? 

```python
>>> np.dtype('complex')
dtype('complex128')
```

Reply:
Fixed - let me know if you think the way I'm handling unsupported NumPy types is OK